File: Projects/MultiFileReader/modules/csv_parser_test.py
#
# Nimish Nayak
# 10-06-2017
#

#
# Csv Parser - Class Tests   
#

from csv_parser import CsvParser
import os

test_file = os.path.join(os.getcwd(),"Test.csv")

def test_csv_parser_with_missing_values():
    
    global test_file    

    # value missing
    with open(test_file,"w")as f:
        f.write("""name,active,value
John,true,
Mark,true,
Paul,false,100
Ben,true,150
""")

    y = CsvParser(test_file)
    y.load_file()
    y.parse_file()
    y.process_file()
    assert(y.get_value() == 150)

def test_csv_parser_with_non_csv_values():
    
    global test_file    

    # value missing
    with open(test_file,"w")as f:
        f.write("""name,active,value
John true 
Mark true 
Paul false 100
Ben true 150
""")

    y = CsvParser(test_file)
    y.load_file()
    y.parse_file()
    y.process_file()
    assert(y.get_value() == 0)

def test_csv_parser_with_blank_file():
    
    global test_file    

    # value missing
    with open(test_file,"w")as f:
        f.write("")

    y = CsvParser(test_file)
    y.load_file()
    y.parse_file()
    y.process_file()
    assert(y.get_value() == 0)

def test_csv_parser_with_no_child_nodes():
    
    global test_file    

    # value missing
    with open(test_file,"w")as f:
        f.write("""name,active,value""")

    y = CsvParser(test_file)
    y.load_file()
    y.parse_file()
    y.process_file()
    assert(y.get_value() == 0)

    # delete the test file as not required 
    os.remove(test_file)

File: Projects/MultiFileReader/modules/csv_parser_test.php
<?php

#
# Nimish Nayak
# 05-07-2017
#

# Test Project to test the code base  
# Functional Test cases

# Note: Developed using PHP 7.0.18-0ubuntu0.16.04.1 (cli) ( NTS )

require_once("./modules/csv_parser.php");
require_once("./modules/utils.php");

class CsvParserTest extends PHPUnit_Framework_TestCase
{
    # Tests 

    public function testParser() {
        
        $testFile1 = Utils::joinPaths(getcwd(),"testdata","test_correct.csv"); 
        $this->csvParser = new CsvParser($testFile1);
        # get the list of dictinary of the yml contents 
        $this->csvParser->parseFile();
        # process the value 
        $this->csvParser->processFile();
        # get the processed value from the parser as a string
        $this->data = $this->csvParser->getValue();              
        $this->assertEquals($this->data, "900");

    }

    public function testParserWithIncorrectYaml() {
        
        $testFile2 = Utils::joinPaths(getcwd(),"testdata","test_incorrect.csv"); 
        $this->csvParser = new YmlParser($testFile2);
        # get the list of dictinary of the yml contents 
        $this->csvParser->parseFile();
        # process the value 
        $this->csvParser->processFile();
        # get the processed value from the parser as a string
        $this->data = $this->csvParser->getValue();              
        $this->assertEquals($this->data, 0);
    }

}

?>